atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/cache/ValidatingTransactionScopedCache.java

```diff
@@ -64,20 +65,45 @@ public void delete(TableReference tableReference, Set<Cell> cells) {
     @Override
     public Map<Cell, byte[]> get(
             TableReference tableReference,
-            Set<Cell> cell,
+            Set<Cell> cells,
             BiFunction<TableReference, Set<Cell>, ListenableFuture<Map<Cell, byte[]>>> valueLoader) {
-        if (shouldValidate.get()) {
-            Map<Cell, byte[]> remoteReads = AtlasFutures.getUnchecked(valueLoader.apply(tableReference, cell));
-            Map<Cell, byte[]> cacheReads =
-                    delegate.get(tableReference, cell, (table, cells) -> getCells(remoteReads, cells));
+        return AtlasFutures.getUnchecked(getAsync(tableReference, cells, valueLoader));
+    }
+
+    @Override
+    public ListenableFuture<Map<Cell, byte[]>> getAsync(
```

Comment:
Exposing this actually turned out to be fairly gnarly. I'm pretty convinced this is correct, but a really tight sanity check or suggestions for further tests would be welcome!